lib/ts/nextjs.ts

```diff
@@ -127,6 +128,219 @@ export default class NextJS {
             });
         };
     }
+
+    private static async commonSSRSession(
+        baseRequest: PreParsedRequest,
+        options?: VerifySessionOptions,
+        userContext?: Record<string, any>
+    ): Promise<{
+        session: SessionContainer | undefined;
+        hasToken: boolean;
+        hasInvalidClaims: boolean;
+        baseResponse: CollectingResponse;
+        nextResponse?: Response;
+    }> {
+        let baseResponse = new CollectingResponse();
+
+        try {
+            let session = await Session.getSession(baseRequest, baseResponse, options, userContext);
+            return {
+                session,
+                hasInvalidClaims: false,
+                hasToken: session !== undefined,
```

Comment:
hasToken should be based on if the request actually contained the token and not based on `session !== undefined`. Cause the getSession can still return session as undefined even though the token exists, in case the user has revoked the access token offline, and has passed in `checkDatabase: true` && `sessionRequired: false` in the `options`.

Please double check this logic and make sure you have covered all the cases properly... + add tests for this

Reply:
done